### PR TITLE
add `attackWhileBlocking` and `attackWhileBreaking` to AimAssist | `attackWhileBreaking` to TriggerBot

### PR DESCRIPTION
**Pull request by TejasLamba2006**
## Description
[Click here](https://github.com/Wurst-Imperium/Wurst7/pull/947#issuecomment-1930490549)

## Summary by CodeRabbit

- **New Features**
	- Enhanced Aim Assist functionality with options for attacking while blocking with a shield or breaking blocks, improving combat and mining activities.
	- Expanded Trigger Bot capabilities to include attacking enemies while breaking blocks, offering increased flexibility during gameplay.

### CODE REVIEW

**Comment by TejasLamba2006** [comment]
@coderabbitai generate a summary with the changes

File: src/main/java/net/wurstclient/hacks/AimAssistHack.java [hunk 1]
```diff
@@ -48,6 +49,15 @@ public final class AimAssistHack extends Hack
 	private final CheckboxSetting checkLOS = new CheckboxSetting(
 		"Check line of sight", "Won't aim at entities behind blocks.", true);
 	
+	private final CheckboxSetting attackWhileBlocking = new CheckboxSetting(
+		"Attack while blocking",
+		"Whether or not to attack while blocking with a shield / using items.",
+		false);
```
[the remark this answers is not in the document]
Comment (TejasLamba2006) [reply src/main/java/net/wurstclient/hacks/AimAssistHack.java:55]: could you make me a description for `attackWhileBlocking` that is clear and concise for end-users.

File: src/main/java/net/wurstclient/hacks/AimAssistHack.java [hunk 2]
```diff
@@ -48,6 +49,15 @@
...
 		"Check line of sight", "Won't aim at entities behind blocks.", true);
 	
+	private final CheckboxSetting attackWhileBlocking = new CheckboxSetting(
+		"Attack while blocking",
+		"Whether or not to attack while blocking with a shield / using items.",
+		false);
+	
+	private final CheckboxSetting attackWhileBreaking =
+		new CheckboxSetting("Attack while breaking",
+			"Whether or not to attack while breaking blocks.", false);
```
[the remark this answers is not in the document]
Comment (TejasLamba2006) [reply src/main/java/net/wurstclient/hacks/AimAssistHack.java:59]: make me a description for `attackWhileBreaking` that is clear and concise for end-users.

File: src/main/java/net/wurstclient/hacks/TriggerBotHack.java [hunk 3]
```diff
@@ -37,6 +37,10 @@ public final class TriggerBotHack extends Hack implements UpdateListener
 		"Whether or not to attack while blocking with a shield / using items.",
 		false);
 	
+	private final CheckboxSetting attackWhileBreaking =
+		new CheckboxSetting("Attack while breaking",
+			"Whether or not to attack while breaking blocks.", false);
```
[the remark this answers is not in the document]
Comment (TejasLamba2006) [reply src/main/java/net/wurstclient/hacks/TriggerBotHack.java]: I pushed a fix in commit 8f48ff7

File: src/main/java/net/wurstclient/hacks/AimAssistHack.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (TejasLamba2006) [reply src/main/java/net/wurstclient/hacks/AimAssistHack.java]: I pushed a fix in commit fafea3f

File: src/main/java/net/wurstclient/hacks/AimAssistHack.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (TejasLamba2006) [reply src/main/java/net/wurstclient/hacks/AimAssistHack.java]: I pushed a fix in commit fafea3f

**Review by Alexander01998 — CHANGES_REQUESTED** [review]
Hey @TejasLamba2006, thanks for the PR!
Seems like a useful addition and the code looks good overall. 👍 
I found a few minor issues though. Do you have time to address those?

**Comment by TejasLamba2006** [comment]
Hey @Alexander01998, did what you asked.

**Review by Alexander01998 — APPROVED** [review]

**Comment by Alexander01998** [comment]
Upon further testing, unfortunately the "attack/aim while breaking" settings don't work when using other hacks like Nuker to break the blocks. I will only be merging the "attack while blocking" checkbox for now.

**Comment by TejasLamba2006** [comment]
@coderabbitai configuration

**Pull request merged**